backend/packages/Upgrade/src/api/services/ExperimentAssignmentService.ts

```diff
@@ -349,54 +338,96 @@ export class ExperimentAssignmentService {
           }
         }
       }
-      // adding in monitored experiment point table
-
-      const assignmentUnit = experiments
-        ? experiments.find((exp) => exp.id === experimentId)?.assignmentUnit || experiments[0]?.assignmentUnit
-        : null;
-      monitoredDocument = await this.monitoredDecisionPointRepository.saveRawJson({
-        id: monitoredDocument?.id || uuid(),
-        experimentId: experimentId,
-        condition: assignmentUnit === ASSIGNMENT_UNIT.WITHIN_SUBJECTS ? null : condition,
-        user: userDoc,
-        site: site,
-        target: target,
-      });
-
-      // save monitored log document
-      const logDocument = await this.monitoredDecisionPointLogRepository.save({
-        monitoredDecisionPoint: monitoredDocument,
-        condition: condition,
-        uniquifier: uniquifier,
-      });
-      return {
-        ...monitoredDocument,
-        condition: monitoredDocument.condition || logDocument.condition,
-      };
     } else {
-      const assignmentUnit = experiments
-        ? experiments.find((exp) => exp.id === experimentId)?.assignmentUnit || experiments[0]?.assignmentUnit
-        : null;
-      const monitoredDocument = await this.monitoredDecisionPointRepository.saveRawJson({
-        id: uuid(),
-        experimentId: experimentId,
-        condition: assignmentUnit === ASSIGNMENT_UNIT.WITHIN_SUBJECTS ? null : condition,
-        user: userDoc,
+      // experiment level exclusion
+      const experiment = experiments[0];
```

Comment:
This else case means there is no valid experiment so I am not sure how this code will work here. Ideally, `filteredExperiment` should be an empty array here.